### ISSUE TITLE
Not working on Debian Buster (.deb and .AppImage)

### ISSUE THREAD

**Issue by starfetch** [music-assistant/companion#11]
I tried to get the Music Assistant Companion v0.0.22 (great piece of software, btw!) to run on a Debian Buster machine. 
With the `.deb` package, it installs (seemingly) fine with `apt install`, but trying to run it with `music-assistant-companion` (a recognised command), results in the following error:

```bash 
music-assistant-companion: error while loading shared libraries: libssl.so.3: 
cannot open shared object file: No such file or directory
```

Trying the `.AppImage`, after making it executable, results in the following error(s):
```bash
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by music-assistant-companion)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by music-assistant-companion)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.29' not found (required by music-assistant-companion)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by music-assistant-companion)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by music-assistant-companion)
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.35' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libwebkit2gtk-4.0.so.37)
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libwebkit2gtk-4.0.so.37)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libwebkit2gtk-4.0.so.37)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libwebkit2gtk-4.0.so.37)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libwebkit2gtk-4.0.so.37)
music-assistant-companion: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libwebkit2gtk-4.0.so.37)
music-assistant-companion: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libwebkit2gtk-4.0.so.37)
music-assistant-companion: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libwebkit2gtk-4.0.so.37)
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libgtk-3.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libgtk-3.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libgdk-3.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libcairo.so.2)
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libcairo.so.2)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libgio-2.0.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libgio-2.0.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libjavascriptcoregtk-4.0.so.18)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libjavascriptcoregtk-4.0.so.18)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libjavascriptcoregtk-4.0.so.18)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libjavascriptcoregtk-4.0.so.18)
music-assistant-companion: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libjavascriptcoregtk-4.0.so.18)
music-assistant-companion: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libjavascriptcoregtk-4.0.so.18)
music-assistant-companion: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libjavascriptcoregtk-4.0.so.18)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libglib-2.0.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libglib-2.0.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libglib-2.0.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libcrypto.so.3)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libcrypto.so.3)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libwpe-1.0.so.1)
music-assistant-companion: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libWPEBackend-fdo-1.0.so.1)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libepoxy.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libicui18n.so.70)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.30' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libsystemd.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libsystemd.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libsystemd.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libsystemd.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libicuuc.so.70)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libicuuc.so.70)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libicuuc.so.70)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libxml2.so.2)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libxml2.so.2)
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libxml2.so.2)
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libsqlite3.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libsqlite3.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libsqlite3.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libxslt.so.1)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libxslt.so.1)
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/liblcms2.so.2)
music-assistant-companion: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libwoff2dec.so.1.0.2)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libgcrypt.so.20)
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libgstreamer-1.0.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libgstreamer-1.0.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libgstreamer-1.0.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libgstpbutils-1.0.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libgstaudio-1.0.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libgsttag-1.0.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libgstvideo-1.0.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libpng16.so.16)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libopenjp2.so.7)
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libopenjp2.so.7)
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libwebp.so.7)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libwebp.so.7)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libenchant-2.so.2)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libgmodule-2.0.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libwayland-server.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libwayland-server.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libxkbcommon.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libpixman-1.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libpsl.so.5)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libgssapi_krb5.so.2)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libmount.so.1)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libmount.so.1)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libselinux.so.1)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.30' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libselinux.so.1)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libselinux.so.1)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/liblzma.so.5)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/liblzma.so.5)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libcap.so.2)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libunwind.so.8)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libunwind.so.8)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libdw.so.1)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libdw.so.1)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libgudev-1.0.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libevdev.so.2)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libdbus-1.so.3)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libdbus-1.so.3)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libXau.so.6)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libunistring.so.2)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libunistring.so.2)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libkrb5.so.3)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libkrb5.so.3)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libk5crypto.so.3)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libkrb5support.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libkrb5support.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libblkid.so.1)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libelf.so.1)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libudev.so.1)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libudev.so.1)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.30' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libudev.so.1)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libbsd.so.0)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /tmp/.mount_music-WLBzxg/usr/lib/libmd.so.0)
```

Searching for the error very briefly led me to this [stack answer](https://stackoverflow.com/questions/74740941/how-can-i-resolve-this-issue-libm-so-6-version-glibc-2-29-not-found-c-c/76490135#76490135), which cautions against updating these dependencies, since a lot of the OS depends on it. 

Can I therefore assume, that I am simply out of luck on my old Buster machine? I could not find `requirements` in the `README`, therefore the issue.

Happy to provide more system info if needed, and also to close the issue as a `wontfix` if it is simply my setup being too old.

**Comment by arctixdev** [comment]
Hi @starfetch, im glad you are enjoying Music Assistant. Sorry for not responding till now, I was busy IRL. 

I think the problem is that the deb and appimage builds are being build on Ubuntu 22.04 which is based on Debian 12. I think changing it to Ubuntu 20.04 instead should solve the problem :)

Normally I could release a new version within 15 minutes. But in the last version I rotated the signing keys for the app to fix a vulnerbilty. And sadly I dont have permissions to change the repo environment secrets meaning the new builds wouldnt get accepted by the app updater. And the guy who has the rights to update it hasnt answered me yet...
TLDR: I cant release a new version with a fix. I might be able to create a pre-release or something with the fix for you.

**Comment by arctixdev** [comment]
This version should be fixed: https://github.com/music-assistant/music-assistant-desktop/releases/tag/v0.0.23

**Comment by starfetch** [comment]
Thanks for the lightning fast work! 

But, I'm sorry to say, the errors persist... 
The `.AppImage` error is the same as above, complaining about `GLIBC`, and now the `.deb` package, which still installs without complaint, has also turned to complaining about it:
```bash
music-assistant-companion: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by music-assistant-companion)
music-assistant-companion: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.29' not found (required by music-assistant-companion)
```
So... one step further?

Note: This error is by no means urgent or limiting on my side, since I can run a `squeezelite` client on this Buster machine to connect to MA anyways, so please don't feel obligated to spend more time on it, not sure how many other people would benefit from compatibility with Buster ;)

**Comment by arctixdev** [comment]
Hmm alright, well. Im not sure what the problem is then.. You could try building the app youself. There are some weak instructions in the [README](https://github.com/music-assistant/music-assistant-desktop/releases/tag/v0.0.23)

**Comment by arctixdev** [comment]
Merging with #28
